### ISSUE TITLE
Zoomed in picture?

### ISSUE THREAD

**Issue by anonimno1** [pprevos/denote-explore#21]
Hi @pprevos,

I am using Emacs-29.3 with denote-explore 20240616.359.
Installed using this code from you web page.

>  (use-package denote-explore
    :ensure t
    :custom
      ;; Where to store network data and in which format
      (denote-explore-network-directory "~/myorg/notes/graphs/")
      (denote-explore-network-filename "denote-network")
      (denote-explore-network-format 'graphviz)
    :bind
      (;; Statistics
       ("C-c n e c" . denote-explore-count-notes)
       ("C-c n e C" . denote-explore-count-keywords)
       ("C-c n e b" . denote-explore-keywords-barchart)
       ("C-c n e x" . denote-explore-extensions-barchart)
       ;; Random walks
       ("C-c n e r" . denote-explore-random-note)
       ("C-c n e l" . denote-explore-random-link)
       ("C-c n e k" . denote-explore-random-keyword)
       ;; Denote Janitor
       ("C-c n e d" . denote-explore-identify-duplicate-notes)
       ("C-c n e z" . denote-explore-zero-keywords)
       ("C-c n e s" . denote-explore-single-keywords)
       ("C-c n e o" . denote-explore-sort-keywords)
       ("C-c n e r" . denote-explore-rename-keywords)
       ;; Visualise denote
       ("C-c n e n" . denote-explore-network)
       ("C-c n e v" . denote-explore-network-regenerate)
       ("C-c n e D" . denote-explore-degree-barchart)))

Somehow the pictures of generated networks with Graphviz look strange. Basically they seem to show only a zoomed in part of the description found in the file.
What am I doing wrong?

[denote-network.txt](https://github.com/user-attachments/files/15973865/denote-network.txt)
![image](https://github.com/pprevos/denote-explore/assets/46791481/bac2d7bf-d900-4945-a0fc-98506ad7b412)

**Comment by pprevos** [comment]
Hi @anonimno1,

That is indeed strange behaviour. 

Some tests:
1. Can you open `~/myorg/notes/graphs/denote-network.svg` with other software? Perhaps it is the browser.
2. If that gives the same result, then copy the content of `~/myorg/notes/graphs/denote-network.dot` to [Neato](https://edotor.net/)

**Comment by anonimno1** [comment]
So I opened the svg in Chrome and Gimp, the same thing. BUT when I paste the gv-content to Neato, I get a full and nice picture. 
denote-explore generates the gv-text-file, but graphviz creates the svg from it, right? Could graphviz be the problem? I am using  2.42.2-6 from Ubuntu 22.04 LTS.

**Comment by anonimno1** [comment]
So I pulled the latest GraphViz from https://gitlab.com/graphviz/graphviz/-/releases, but didn't solve the issue.

**Comment by pprevos** [comment]
That's weird. I use GraphViz 11.0.0-1 on Arch Linux.

I checked using GraphViz directly with `dot denote-network.gv -Tsvg -o test.svg` and it worked for all three types.

If you don't mind, can you share me the link to your `gv` source code inside https://edotor.net/ (which I mistakenly called Neato).

**Comment by anonimno1** [comment]
Hi @pprevos ,

thank you for giving me the code to test directly with GraphViz, that led me to the solution. It seems GraphViz in Ubuntu 22.04 doesn't work with denote-explore. As I first got the v11, I upgraded to it. This brought no change. But once I completly purged GraphViz and all dependacies, then installed v11 again ... it works now :)!
Great and thank you again.

**Comment by anonimno1** [comment]
Problem in GraphViz from Ubuntu 22.04.

**Comment by lucasgruss** [comment]
Hi, 
I would like to further the discussion, as I have encountered the same problem with denote-explore and graphviz 2.43.0 on Debian 12.6. Opening the generated svg file in inkscape gives an idea of what is going on: the actual graph is larger than the size of the image, see below on the small graph I generated (the black rectangle is the actual image and is the area that will be displayed image viewers).
![image](https://github.com/user-attachments/assets/4e300f77-bcf1-4a35-a19a-6a39ec13244e)
Manually expanding the image properties to be large enough fixed it in the browser.

I am unsure what could be done to ensure that the size of the image is large enough to contain all of the graph, there might be a command line option that can be used to make this work for all versions of graphviz. I will look into it if I can spend time on it. In the mean time, I upgraded to the latest version of graphviz (12.1.0) which fixed the issue for me too.
